pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -197,7 +197,7 @@ def format_checkpoint_name(self, epoch, metrics, ver=None):
         return filepath
 
     @rank_zero_only
-    def on_validation_end(self, trainer, pl_module):
+    def on_epoch_end(self, trainer, pl_module):
```

Comment:
Hi! Does this change effect checkpointing in the middle of training epoch? Consider the usecase where we train on a large dataset and we want to checkpoint & early stop every X steps instead of every X epoches, for example X = 100, i.e. `val_check_interval` = 100.